### ISSUE TITLE
China Helix and USA Chinook repair at half rate

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#745]
China Helix and USA Chinook and USA Combat Chinook repair at half rate on an Airfield.

```
Armor ChinookArmor
  Armor = DEFAULT           50%  ;this sets the level for all nonspecified damage types
```

**Comment by Jundiyy** [comment]
I think add the 50% code to the Armour and leave as is. Giving them 100% repair rate might be too powerful.

**Comment by ExiLe-ZH** [comment]
Lix feels painfully slow tbf, some sort of buff wouldn't hurt. Would help vs GLA that can instantly heal all it's units. 

Nooks are rarely repaired by players, it's not worth to build an airfield for this purpose, neither with a buff. However when you already have an airfield it might be an option to repair it if it's faster. Currently it's just better to make a new Nook.

How do they compare to Comanche healing exactly though?

**Comment by commy2** [comment]
Related: [GLA Structures repair at 75% with Fortified Structure upgrade](https://github.com/TheSuperHackers/GeneralsGamePatch/issues/228)
[Fix: Make Reduced Repair Rate On Fortified Structures Explicit](https://github.com/TheSuperHackers/GeneralsGamePatch/pull/246)

Also: This also happens to AFG Comanches, but only after the Countermeasures upgrade. They heal at 90% rate. Only AFG, not the other USA sub-factions.

**Comment by Jundiyy** [comment]
As mentioned on Discord, 100% would be too much and make it quite powerful. But having 100% might mean we don't need any other changes to China army since it would make Helix stronger, faster to repair and reuse. 

Perhaps a small increase would be okay.

Chinooks don't need to be changes imo, perhaps make a copy of the armour and give Helix the edited one.

**Comment by xezon** [comment]
I doubt a faster Helix repair will buff China so much that it is on par with USA and GLA. Airforce will just send one King Raptor that penetrates 20 Rocket men and kill it right in the air when the Helix repair is done.

**Comment by Jundiyy** [comment]
That is true, but in games where China spams Helices, this would become even better. But, thinking over it, perhaps maybe that buff is needed. Maybe it wouldn't overdo it.

**Comment by commy2** [comment]
Note that repair ports in War Factories and Airfields work differently in general.

For War Factories, any unit is repaired by 20% of their total health per second.
Airfields on the other hand heal an absolute amount. They heal 10 health points per second.

Health points vary widely between units. A MIG has 160 health, a Helix has 300 health.
While it will take a Technical and an Overlord exactly 5 seconds each to heal from 1% to 100%, it will, **even after a fix**, take about twice as long to repair a Helix than to repair a MIG.

This is also why the Helix is repaired so slowly: it combines a high health on an absolute repair station with a 50% penality to healing.

**Comment by xezon** [comment]
The China Helix flies relatively slow, so the majority of time is wasted by flying damaged Helix back home. I think China Inf would benefit most from reduced repair time, because they have no Propaganda on Helix.

**Comment by commy2** [comment]
Propaganda/Subliminal Messaging also repair a relative amount. This means that they scale better with veterancy and high health units like Overlords.

**Comment by RisingZH** [comment]
Agree that it's too slow at the moment, and that possibly doubling it could be a little too much (but maybe not, would need to test).

On a similar note, migs without black napalm take soooo long to reload, it's very out of line with all other planes. Is this a bug? That seems like a good change that would make the game more consistent and also help balance china vs usa and gla early game.

**Comment by commy2** [comment]
Mig reload is strange and inconsistent, altough it may be intentional. Before Black Napalm, the reload time is 8 seconds. After Black Napalm, the reload time is 2 seconds.

Same happens for Nuke Migs and the Nuke Mig upgrade.

Should open its own issue for that. For the record, Raptor reload time is also 8 seconds (but for all four missiles instead of two). King Raptor reload time is just 2 seconds (for all six missiles).

**Comment by MTKing4** [comment]
> I think add the 50% code to the Armour and leave as is.

What do you mean by "add the code to the armour" ? What would that do to buff the helix? I don't understand

**Comment by Jundiyy** [comment]
> > I think add the 50% code to the Armour and leave as is.
> 
> What do you mean by "add the code to the armour" ? What would that do to buff the helix? I don't understand

That part doesn't do anything, just makes it more clear in code. (That's if we leave it at 50%.)

**Comment by commy2** [comment]
> What would that do to buff the helix?

Nothing, it would document the current behaviour as intended.

**Comment by MTKing4** [comment]
I see, nah we definitely should buff it then

**Comment by xezon** [comment]
Helipad heals at 10 health per second when HEALING = 100%

**Comment by xezon** [comment]
![helihealing2](https://user-images.githubusercontent.com/4720891/182943430-47095374-51f6-4260-ad51-c9f98917bd67.png)

**Comment by xezon** [comment]
Default Helix Repair time is long with 60 seconds, however, paired with non upgraded Propaganda Tower it will drop to comfortable 27 seconds, which is very competitive with Comanche. A Subliminal Propaganda Tower will even repair quicker, which is not shown in above chart, as it does not show the 4% Heal Percent Stat.

We submitted change to make China Propaganda Tower available before Propaganda Center. This will make a boosted Repair Time more accessible if the player invests 500 money for it.

As for USA, since it does not need buffs to be competitive, a decreased Chinook repair time is not necessary. A reduction would potentially be detrimental to balancing regarding Airforce Combat Chinook.

There is no evidence that we need to tweak the Healing of Helicopters.
